core/client/src/main/java/alluxio/client/lineage/AbstractLineageClient.java

```diff
@@ -63,12 +65,19 @@ public long createLineage(List<AlluxioURI> inputFiles, List<AlluxioURI> outputFi
     // TODO(yupeng): relax this to support other type of jobs
     Preconditions.checkState(job instanceof CommandLineJob,
         PreconditionMessage.COMMAND_LINE_LINEAGE_ONLY);
-    LineageMasterClient masterClient = mContext.acquireMasterClient();
+    LineageMasterClient masterClient = null;
     try {
+      masterClient = mContext.acquireMasterClient();
       long lineageId = masterClient.createLineage(stripURIList(inputFiles),
           stripURIList(outputFiles), (CommandLineJob) job);
       LOG.info("Created lineage {}", lineageId);
       return lineageId;
+    } catch (NotFoundException e) {
+      throw new FileDoesNotExistException(e.getMessage());
+    } catch (UnavailableException e) {
+      throw new IOException(e.getMessage());
+    } catch (AlluxioStatusException e) {
+      throw e.toAlluxioException();
```

Comment:
what I am pointing out is that there does not seem to be a consistent pattern for what to convert `AlluxioStatusException` to ... which means that it will be inconsistent

I assume this inconsistency will go away in 2.0 when we will be able to change the user API, correct?